src/OpenTracing.Contrib.NetCore/CoreFx/SqlClientDiagnostics.cs

```diff
@@ -28,7 +28,8 @@ internal sealed class SqlClientDiagnostics : DiagnosticListenerObserver
         protected override void OnNext(string eventName, object untypedArg)
         {
             switch (eventName)
-            {
+            { 
+                case "Microsoft.Data.SqlClient.WriteCommandBefore":
                 case "System.Data.SqlClient.WriteCommandBefore":
                     {
                         var args = (SqlCommand)_activityCommand_RequestFetcher.Fetch(untypedArg);
```

Comment:
Thank you for the PR! I'm not sure if this will work though, because this currently always casts to `System.Data.SqlClient.SqlCommand` (see `using`-statements). The new `Microsoft.Data.SqlClient`-package introduces new namespaces so any event from that package would pass a `Microsoft.Data.SqlClient.SqlCommand` and therefore result in failing cast. 

I guess we would need either a completely separate class that only handles the `Microsoft.Data.SqlClient`-events (to have a clean separation) or different logic in the `Microsoft.Data.SqlClient.WriteCommandBefore`-case.

If you want this merged, you'd have to look into this please. I'm keeping my efforts on this project to a minimum as OpenTelemetry for .NET has reached RC1 recently and therefore will soon make this project obsolete.

Reply:
I just tested this and the casts fail, so I can't merge this. I will however create a different PR with separate diagnostic listeners.